File: 2144-maximum-difference-between-increasing-elements/maximum-difference-between-increasing-elements.java
class Solution {
    public int maximumDifference(int[] nums) {
        int min = nums[0], max = 0;
        for (int i = 1; i < nums.length; i++)
            if (nums[i] < min)
                min = nums[i];
            else
                max = max > nums[i] - min ? max : nums[i] - min;

        return max != 0 ? max : -1;
    }
}

File: 2144-maximum-difference-between-increasing-elements/maximum-difference-between-increasing-elements.go
func maximumDifference(nums []int) int {
    min, dif := nums[0], 0;
    for i := 1; i < len(nums); i++ {
        if min > nums[i] {
            min = nums[i];
        } else {
            dif = max(dif, nums[i] - min);
        }
    }

    if dif > 0 {
        return dif;
    }

    return -1;
}